packages/batch-tx-handler/src/l1-batch-handler/utils.ts

```diff
@@ -85,6 +89,42 @@ export const getAllL2Msgs = (l2segments: Uint8Array[]): Uint8Array[] => {
   return l2Msgs;
 };
 
+const extractL2Msg = async (
+  l2segments: Uint8Array[],
+  delayedMessageIndex: number,
+  index: number,
+): Promise<[Uint8Array[], number]> => {
+  let l2Msgs: Uint8Array[] = [];
+  let currentDelayedMessageIndex = delayedMessageIndex;
+  if (index < l2segments.length - 1) {
```

Comment:
Are you traversing the l2segments array from the end to the beginning?
If so, would an inverted for loop work here? Something like
`for (let i = l2segments.length; i > 0; i--) ...`
or similar.
With the idea of avoiding the recursive calls.